# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/trunk/webrtc/modules/desktop_capture/x11/x_atom_cache.h
# Commit: 3f0981f37c1c
# Full Hash: 3f0981f37c1c499ad52016e526239b390703ba2a
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2019-04-17 21:47:29
# Regressor Bug: 1496359
# File Overlap Count: 1
# Description:
#   Bug 1496359 - Add PipeWire support to desktop capture; r=ng
#   
#   This is an import of upstream commit 318da51f99f91e3de1192f29d7f1824958f9f13e.
#   
#   Depends on D27367
# ==============================================================================

diff -r ceed4e0b3373 -r 3f0981f37c1c media/webrtc/trunk/webrtc/modules/desktop_capture/x11/x_atom_cache.h
--- a/media/webrtc/trunk/webrtc/modules/desktop_capture/x11/x_atom_cache.h	Mon Apr 15 17:12:31 2019 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,43 +0,0 @@
-/*
- *  Copyright (c) 2017 The WebRTC project authors. All Rights Reserved.
- *
- *  Use of this source code is governed by a BSD-style license
- *  that can be found in the LICENSE file in the root of the source
- *  tree. An additional intellectual property rights grant can be found
- *  in the file PATENTS.  All contributing project authors may
- *  be found in the AUTHORS file in the root of the source tree.
- */
-
-#ifndef MODULES_DESKTOP_CAPTURE_X11_X_ATOM_CACHE_H_
-#define MODULES_DESKTOP_CAPTURE_X11_X_ATOM_CACHE_H_
-
-#include <X11/Xlib.h>
-#include <X11/Xatom.h>
-
-namespace webrtc {
-
-// A cache of Atom. Each Atom object is created on demand.
-class XAtomCache final {
- public:
-  explicit XAtomCache(::Display* display);
-  ~XAtomCache();
-
-  ::Display* display() const;
-
-  Atom WmState();
-  Atom WindowType();
-  Atom WindowTypeNormal();
-
- private:
-  // If |*atom| is None, this function uses XInternAtom() to retrieve an Atom.
-  Atom CreateIfNotExist(Atom* atom, const char* name);
-
-  ::Display* const display_;
-  Atom wm_state_ = None;
-  Atom window_type_ = None;
-  Atom window_type_normal_ = None;
-};
-
-}  // namespace webrtc
-
-#endif  // MODULES_DESKTOP_CAPTURE_X11_X_ATOM_CACHE_H_